site/src/pages/release-notes/1.31.0.mdx

```diff
@@ -0,0 +1,137 @@
+---
+date: 2024-11-13
+---
+
+## 🌟 New features
+
+- **Dynamic TLS Configuration**: You can now set TLS configurations dynamically using <type://TlsProvider> #5033 #5228
+  - You can create a <type://TlsProvider> which specifies <type://TlsKeyPair> dynamically.
+    ```java
+    // return a TlsKeyPair for host patterns
+    TlsProvider tlsProvider = TlsProvider
+      .builder()
+      .keyPair("*.a.com", TlsKeyPair.of(...))
+      .keyPair("*.b.com", TlsKeyPair.of(...))
+      .build();
+
+    // return a TlsKeyPair dynamically
+    TlsProvider tlsProvider = hostname -> TlsKeyPair.of(...);
+    ```
+  - The <type://TlsProvider> can be used for both server and clients.
+    ```java
+    // client-side
+    ClientFactory
+      .builder()
+      .tlsProvider(tlsProvider);
+
+    // server-side
+    Server
+      .builder()
+      .tlsProvider(tlsProvider);
+    ```
+- **Access to Raw Request Path**: You can access the raw request path via <type://ServiceRequestContext#rawPath()>. #5931 #5932
+  ```java
+  ServiceRequestContext ctx = ...;
+  String rawPath = ctx.rawPath();
+  ```
```

Comment:
```suggestion
- **Access to Raw Request Path**: You can access the raw request path via <type://ServiceRequestContext#rawPath()>. #5931 #5932
  - This can be useful when you need to get the original request path sent from the client, even if its potentially insecure.
    ```java
    ServiceRequestContext ctx = ...;
    String rawPath = ctx.rawPath();
    // rawPath may contain unencoded, invalid or insecure characters.
    ```
```